server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -1191,10 +1195,20 @@ public void evaluateOnDemandTabletsForUnload() {
       UnloaderParams params = new UnloaderParamsImpl(tid, new ServiceEnvironmentImpl(context),
           subset, onDemandTabletsToUnload);
       unloaders.get(tid).evaluate(params);
-      try (TabletsMutator tm = getContext().getAmple().mutateTablets()) {
+      try (var tabletsMutator = getContext().getAmple().conditionallyMutateTablets()) {
         onDemandTabletsToUnload.forEach(ke -> {
           log.debug("Unloading on-demand tablet: {} for table: {}", ke, tid);
-          tm.mutateTablet(ke).deleteHostingRequested().mutate();
+          tabletsMutator.mutateTablet(ke).requireLocation(myLocation).deleteHostingRequested()
+              .submit(tm -> !tm.getHostingRequested());
+        });
+
+        tabletsMutator.process().forEach((extent, result) -> {
+          if (result.getStatus() != Ample.ConditionalResult.Status.ACCEPTED) {
+            var loc = Optional.ofNullable(result.readMetadata()).map(TabletMetadata::getLocation)
+                .orElse(null);
+            log.debug("Failed to clear hosting request marker for {} location in metadata:{}",
+                extent, loc);
```

Comment:
```suggestion
        if (!loc.equals(myLocaion) {
            log.debug("Failed to clear hosting request marker for {} location in metadata:{}",
                extent, loc);
        }
```
I wonder if we should only print this message if the location is *this* location. I guess in theory it's possible that the tablet was unloaded and re-hosted on another tablet server?

Reply:
> I guess in theory it's possible that the tablet was unloaded and re-hosted on another tablet server?

Yeah this runs outside of the tablet code so its possible that the tablet unloads between the check of tablet status and writing the conditional mutation.  So this log message could be noise and the suggested change would probably be good to do later on after testing more extensively.